contracts/GoodGhosting.sol

```diff
@@ -136,6 +154,20 @@ contract GoodGhosting is Ownable, Pausable {
         _unpause();
     }
 
+    /**
+       Allowing the admin to withdraw the pool fees
+    */
+    function adminFeeWithdraw() external onlyOwner whenGameIsCompleted {
+        require(!adminWithdraw, "Admin has already withdrawn");
+        require(adminFee > 0, "No Fees Earned");
+        adminWithdraw = true;
+        require(
+            IERC20(daiToken).transfer(owner(), adminFee),
+            "Fail to transfer ER20 tokens to admin"
+        );
+        emit AdminWithdrawal(owner(), totalGameInterest, adminFee);
```

Comment:
Just keeping our pattern of emitting events first (Check-effects-interactions pattern).

```suggestion
        emit AdminWithdrawal(owner(), totalGameInterest, adminFee);
        require(
            IERC20(daiToken).transfer(owner(), adminFee),
            "Fail to transfer ER20 tokens to admin"
        );
```